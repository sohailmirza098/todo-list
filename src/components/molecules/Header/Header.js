import React from 'react';

import { Toolbar, AppBar, Typography } from '@material-ui/core/'
function Header() {
    return (
        <AppBar position="static">
            <Toolbar>
            <Typography variant="h1" >
                TodoList
            </Typography>
        </Toolbar>
        </AppBar>
    )
}

export default Header;