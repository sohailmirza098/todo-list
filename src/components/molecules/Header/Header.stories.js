import React from 'react';
import Header from "./Header";
import { action } from '@storybook/addon-actions';
// import { text } from '@storybook/addon-knobs/react';

export default {
  title: 'Header',
  component: Header,
};

export const input = () => {
    return(<Header />);
};
