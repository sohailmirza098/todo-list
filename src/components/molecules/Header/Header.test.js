import React from "react";
import { create } from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Header from "./Header";

describe("Header testing", () => {

  test("Matches the snapshot", () => {
    const input = create(<Header />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("Header is present", () => {
    const {getByTestId}=  render(<Header />);
    const header = getByTestId("headerid");
    expect(header).toBeInTheDocument();
  });

  test("Header text content", () => {
    const {getByTestId}=  render(<Header />);
    const header = getByTestId("headerid");
    expect(header.textContent).toEqual("TodoList")
  });

});