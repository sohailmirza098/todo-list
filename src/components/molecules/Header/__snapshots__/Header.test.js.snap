exports[`Header testing Matches the snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  data-testid="headerid">
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters">
    <h1
      className="MuiTypography-root MuiTypography-h1">
      TodoList
    </h1>
  </div>
</header>
`;
